src/library/librarywatcher.h

```diff
@@ -200,11 +217,27 @@ class LibraryWatcher : public QObject {
    */
   QStringList best_image_filters_;
 
-  bool stop_requested_;
   bool scan_on_startup_;
   bool monitor_;
 
-  QMap<int, Directory> watched_dirs_;
+  typedef QMap<int, WatchedDir> WatchListBase;
+  // All methods of QMap are reentrant We should only need to worry about
+  // syncronizing methods that remove directories on the watcher thread and
+  // methods that modify directories called from other threads.
+  class WatchList : public WatchListBase {
```

Comment:
Not really a fan of inheriting from collection classes either. This is much more readable as an object containing a map and a mutex. The typedef is unnecessary too.

Reply:
Updated.